controllers/disruption_controller.go

```diff
@@ -189,20 +189,17 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 				return ctrl.Result{}, fmt.Errorf("error updating disruption injection status: %w", err)
 			}
 
-			isCleaned, err := r.cleanDisruption(instance)
+			r.Recorder.Event(instance, "Normal", "DurationOver", fmt.Sprintf("The disruption has lived longer than its specified duration, and will be garbage collected after %s.", r.ExpiredDisruptionGCDelay))
+
+			chaosPods, err := r.getChaosPods(instance, nil)
 			if err != nil {
+				r.log.Errorw("couldn't list instance's chaos pods", "err", err)
+
 				return ctrl.Result{}, err
 			}
 
-			if !isCleaned {
-				requeueAfter := time.Duration(rand.Intn(5)+5) * time.Second //nolint:gosec
-
-				r.log.Infow(fmt.Sprintf("disruption has not been fully cleaned yet, re-queuing in %v", requeueAfter))
-
-				return ctrl.Result{
-					Requeue:      true,
-					RequeueAfter: requeueAfter,
-				}, r.Update(context.Background(), instance)
+			for _, pod := range chaosPods {
```

Comment:
Aren't we gonna flood someone having a disruption with a lot of chaos pods here?

Reply:
We aren't notifying for every event on a chaos pod, are we? These should only show up if you describe the chaos pods

Reply:
Only on warning events, nvm